djangoplicity/pages/tests/test_models.py

```diff
@@ -0,0 +1,150 @@
+
+        page.start_publishing = datetime.now() + timedelta(days=1)
+        self.assertFalse(page.is_online())
+        page.start_publishing = None
+
+        page.end_publishing = datetime.now() - timedelta(days=1)
+        self.assertFalse(page.is_online())
+        page.end_publishing = None
+
+        # translatedPage = PageFactory.create()
+        # translatedPage.source = PageFactory.create()
+        
+        # self.assertTrue(translatedPage.is_online())
+
+    def test_get_absolute_url(self):
+        translatedPage = PageFactory.create()
+        translatedPage.source = PageFactory.create()
+        self.assertIsNotNone(translatedPage.get_absolute_url())
+
+        translatedPage.source = None
+        self.assertIsNotNone(translatedPage.get_absolute_url())
+
+
+    def test_save(self):
+        section = SectionFactory.build()
+        section.save()
+        page = PageFactory.build()
+        page.script = '<script></script>'
+        page.section = section
+        page.save()
+
+        self.assertFalse(page.live_edit)
+
+    def test_test_render_errors(self):
+        page = PageFactory.build()
+        page.test_render_errors()
+
+
+class URLTest(TestCase):
+    def test_unicode_representation(self):
+        url = URLFactory.build()
+        self.assertEqual(u'%s' % url, 'http://someurl.com') 
+
+class PageGroupTest(TestCase):
+    def test_unicode_representation(self):
+        page_group = PageGroupFactory.build()
+        self.assertEqual(u'%s' % page_group, 'TestingName')
+
+class EmbeddedPageKeyTest(TestCase):
+    def test_unicode_representation(self):
+        page_key = EmbeddedPageKeyFactory.build()
+        self.assertEqual(
+            u'%s' % page_key, u'%s (%s)' % (page_key.title, page_key.page_key))
+
+class RegisterPageKeyTest(TestCase):
+    def test_raise_improperly_configured_except(self):
+        EmbeddedPageKeyFactory.create()
+        self.assertIsNone(register_page_key('testingApp', 'nonExistsKey'))
+
+
+class PageSignalsTest(TestCase):
+    
+    
+    def test_generate_page_id(self):
+        section = SectionFactory.build()
+        section.save()
+        page = PageFactory.build()
+        page.section = section
+        page.save()
+        page.id = None
+
+        self.assertIsNone(generate_page_id(Page, page, None, True))
+        self.assertIsNone(generate_page_id(Page, page, None, False))
+
+
+    def test_page_cache_invalidator(self):
+        section = SectionFactory.build()
+        section.save()
+        page = PageFactory.build()
+        page.section = section
+        page.embedded=True
+        page.url='http://example.com'
+        page.save()
+        self.assertIsNone(page_cache_invalidator(Page, page, None, True))
+        self.assertIsNone(page_cache_invalidator(Page, page, None, False))
+        page.embedded=False
+        new_page = PageFactory.create()
+        new_page.url='http://example.com'
+        new_page.id='123'
+        page.source_id='123'
```

Comment:
You have a bit of formatting errors, are you using an IDE that displays these issues?
```suggestion
        page.url = 'http://example.com'
        page.save()
        self.assertIsNone(page_cache_invalidator(Page, page, None, True))
        self.assertIsNone(page_cache_invalidator(Page, page, None, False))
        page.embedded = False
        new_page = PageFactory.create()
        new_page.url = 'http://example.com'
        new_page.id = '123'
        page.source_id = '123'
```

Reply:
Resolved in [#ad182f52](https://github.com/djangoplicity/djangoplicity/pull/96/commits/ad182f52f0a9dbcfe1655e63e109986108265504)